Dynatrace/dynatrace-operator
# Remove kubebuilder defaults

## Description

[DAQ-1435](https://dt-rnd.atlassian.net/browse/DAQ-1435)

Removes every usage of `// +kubebuilder:default` in the latest versions of the CRDs.

Reasoning:
- These defaults are not always respected, and can cause some weird behaviour.
- Resulting CR is confusing to users 
   - "why did a field in the CR I didn't define?"
- In unittest you have to "imitate" this default setting.
   - because the defaults were enforced by the api-server and not our code

Sidenote:
- There are some "weird" ordering changes in `structs`, this is because of using pointer values for fields that used to have the `// +kubebuilder:default`. (`betteralign -apply ./...` went crazy)
- `address.Of` function is no longer restricted to a random set of types


⚠️ Important change: [Make DynatraceApiRequestThreshold uint16](https://github.com/Dynatrace/dynatrace-operator/pull/4010/commits/40166183a3f9cb6d7f242053a9cb5e5d118cf463)

## How can this be tested?
Deploy DK/EC without setting fields that have defaults, make sure everything works as expected.
- Also check that when overwriting the defaults is also respected.

[DAQ-1435]: https://dt-rnd.atlassian.net/browse/DAQ-1435?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
